Source/Core/AudioCommon/WaveFile.h

```diff
@@ -19,14 +19,18 @@
 
 #include "Common/CommonTypes.h"
 #include "Common/File.h"
-#include "Common/NonCopyable.h"
 
-class WaveFileWriter : NonCopyable
+class WaveFileWriter
 {
 public:
   WaveFileWriter();
   ~WaveFileWriter();
 
+  WaveFileWriter(const WaveFileWriter&) = delete;
+  WaveFileWriter& operator=(const WaveFileWriter&) = delete;
+  WaveFileWriter(WaveFileWriter&&) = delete;
```

Comment:
Why are the move constructors and assignment operators deleted here? Ditto for other similar places.